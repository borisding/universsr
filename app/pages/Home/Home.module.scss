@use 'sass:math';
@import '../../theme/variables';

.lead {
  font-size: 1.25rem;
  font-weight: 300;
}

.logo {
  display: block;
  margin-bottom: math.div($base-margin, 4);
  overflow: hidden;
}
