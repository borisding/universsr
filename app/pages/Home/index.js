import loadable from '@loadable/component';

export default loadable(() => import('./Home'));
