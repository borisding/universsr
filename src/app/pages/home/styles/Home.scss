@import '@common/styles/variables';

.lead {
  font-size: 1.125em;
  font-weight: 400;
}

.logo {
  display: block;
  height: 96px;
  margin-bottom: $base-margin / 4;
  overflow: hidden;
  width: 96px;

  img {
    display: block;
    height: 100%;
    width: 100%;
  }
}
