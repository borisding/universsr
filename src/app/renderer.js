import React from 'react';
import DocumentTitle from 'react-document-title';
import serialize from 'serialize-javascript';
import flushChunks from 'webpack-flush-chunks';
import { renderToString } from 'react-dom/server';
import { flushChunkNames } from 'react-universal-component/server';
import { matchRoutes, renderRoutes } from 'react-router-config';
import { StaticRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import storeFactory from './store';
import routes from './routes';

// preload data for matched route
function prefetchBranchData(store, url) {
  try {
    const branch = matchRoutes(routes, url);
    const promises = branch.map(({ route, match }) => {
      const { loadData } = route;
      const { dispatch } = store;

      if (match && match.isExact && loadData) {
        return Array.isArray(loadData)
          ? Promise.all(loadData.map(action => dispatch(action(match))))
          : dispatch(loadData(match));
      }

      return Promise.resolve(null);
    });

    return Promise.all(promises);
  } catch (err) {
    throw err;
  }
}

// export default server renderer and receiving stats
// also, should allow it to be mounted as middleware for production usage
export default function serverRenderer({ clientStats }) {
  return async (req, res, next) => {
    try {
      const context = {};
      const store = storeFactory();

      await prefetchBranchData(store, req.url);

      const appString = renderToString(
        <Provider store={store}>
          <StaticRouter location={req.url} context={context}>
            {renderRoutes(routes)}
          </StaticRouter>
        </Provider>
      );

      const pageTitle = DocumentTitle.rewind();
      const chunksOptions = { chunkNames: flushChunkNames() };
      const { scripts, stylesheets } = flushChunks(clientStats, chunksOptions);

      const preloadedState = serialize(store.getState(), { isJSON: true });
      const { statusCode = 200, redirectUrl } = context;
      const { nonce } = res.locals;

      if ([301, 302].includes(statusCode) && redirectUrl) {
        return res.redirect(statusCode, redirectUrl);
      }

      res.status(statusCode);
      res.render('index', {
        pageTitle,
        appString,
        preloadedState,
        stylesheets,
        scripts,
        nonce
      });
    } catch (err) {
      next(new Error(err));
    }
  };
}
