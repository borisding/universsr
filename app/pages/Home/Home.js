import React from 'react';
import logo from '../../../assets/images/logo.png';
import { PageContainer } from '../../components';
import styles from './Home.module.scss';

export default function Home() {
  return (
    <PageContainer title="Home">
      <div className={styles.logo}>
        <img src={logo} alt="React Redux Boilerplate" />
      </div>
      <p className={styles.lead}>
        universsr is <strong>universal</strong> React-Redux starter kit.
      </p>
      <br />
      <a
        href="https://github.com/borisding/universsr"
        rel="noopener noreferrer"
        target="_blank"
      >
        GitHub repository
      </a>
    </PageContainer>
  );
}
