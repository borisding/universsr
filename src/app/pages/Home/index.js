import React from 'react';
import Alert from 'react-s-alert';
import logo from '@assets/images/logo.png'; // example of import assets image
import { PageTitle } from '@common/components';
import './styles.scss';

const Home = () => (
  <PageTitle title="Home">
    <div styleName="logo">
      <img src={logo} alt="React Redux Boilerplate" />
    </div>
    <p styleName="lead">
      <span>
        universsr is <strong>universal</strong> React-Redux starter kit.
      </span>
    </p>
    <div>
      <a
        href="#"
        onClick={evt => {
          evt.preventDefault();
          Alert.info('Hello! This is universal React + Redux demo site.');
        }}
      >
        Show alert
      </a>
      {' or, go to '}
      <a href="https://github.com/borisding/universsr">GitHub repository</a>
    </div>
  </PageTitle>
);

export default Home;
