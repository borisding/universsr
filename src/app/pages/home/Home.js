import React from 'react';
import Alert from 'react-s-alert';
import logo from '@assets/images/logo.png'; // example of import assets image
import './styles/Home.scss';

export default () => {
  return (
    <React.Fragment>
      <div styleName="logo">
        <img src={logo} alt="React Redux Boilerplate" />
      </div>
      <p styleName="lead">
        <span>
          universsr is <strong>universal</strong> React-Redux boilerplate.
        </span>
      </p>
      <div>
        <a
          href="#"
          onClick={evt => {
            evt.preventDefault();
            Alert.info('Hello! This is universal React + Redux demo site.');
          }}
        >
          Show alert
        </a>
        {' or, go to '}
        <a href="https://github.com/borisding/universsr">GitHub repository</a>
      </div>
    </React.Fragment>
  );
};
