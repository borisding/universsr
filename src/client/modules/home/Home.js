import React from 'react';
import logo from '@assets/images/logo.png'; // example of import assets image
import './styles/Home.scss';

export default () => {
  return (
    <div>
      <div styleName="logo">
        <img src={logo} alt="React Redux Boilerplate" />
      </div>
      <p styleName="lead">
        <span>
          universsr is <strong>universal</strong> React-Redux boilerplate.
        </span>
        {' | '}
        <a href="https://github.com/borisding/universsr">GitHub Repo</a>
      </p>
    </div>
  );
};
