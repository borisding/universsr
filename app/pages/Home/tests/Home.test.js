import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { HelmetProvider } from 'react-helmet-async';
import Home from '../Home';

describe('Home', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <HelmetProvider>
            <Home />
          </HelmetProvider>
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
