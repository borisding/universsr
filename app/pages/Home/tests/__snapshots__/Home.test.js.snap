// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
Array [
  <div
    className="logo"
  >
    <img
      alt="React Logo"
      src="test-file-stub"
    />
  </div>,
  <p
    className="lead"
  >
    <strong>
      universsr
    </strong>
     is universal React web app boilerplate.
  </p>,
  <br />,
  <a
    href="https://github.com/borisding/universsr"
    rel="noopener noreferrer"
    target="_blank"
  >
    GitHub repository
  </a>,
]
`;
